client/state/console-dispatch/index.js

```diff
@@ -33,7 +33,12 @@ const state = {
 
 const actionLog = {
 	clear: () => ( state.actionHistory = [] ),
-	filter: type => state.actionHistory.filter( matchesProperty( 'type', type ) ),
+	filter: typeTest =>
+		state.actionHistory.filter(
+			typeTest instanceof RegExp
+				? ( { type } ) => typeTest.test( type )
+				: matchesProperty( 'type', typeTest )
+		),
```

Comment:
If we are allowing RegExp objects, which seems like a good idea, then I don't see much need in keeping around `matchedProperty()` and making two types of tests. Might as well just make them all a string test. Maybe even while we're at it, just toss in the towel and allow full searching - why limit it to `type`?

```js
const queryToPredicate = query => {
	if ( query instanceof RegExp ) {
		return ( { type } ) => query.test( type )
	}

	if ( 'string' === typeof query ) {
		return ( { type } ) => type === query;
	}

	if ( 'function' === typeof query ) {
		return query;
	}

	throw new ValueError( 'Can only query by string or RegExp match on `type` or with a given predicate function' );
}

filter: query => state.actionHistory.filter( queryToPredicate( query ) )
```

Reply:
👍  I like it.

Reply:
8c9f3f2